crates/uv-cli/src/lib.rs

```diff
@@ -4081,7 +4081,8 @@ pub struct ToolRunArgs {
     #[arg(long)]
     pub with_editable: Vec<comma::CommaSeparatedRequirements>,
 
-    /// Run with all packages listed in the given `requirements.txt` files.
+    /// Run with all packages listed in the given `requirements.txt` files or PEP 723 Python
+    /// scripts.
     #[arg(long, value_delimiter = ',', value_parser = parse_maybe_file_path)]
     pub with_requirements: Vec<Maybe<PathBuf>>,
```

Comment:
Separately from the documentation, we need to make sure we want `--with-requirements` to support this format. It means we won't be able to read requirements from a PEP 723 script without a `.py` extension (which is more common than I expected) unless we want to start sniffing for the metadata block? Maybe that's reasonable?

Another option is like `--with-script`, which only really applies to `uv run` / `uv tool run` (i.e., we'd need something else in `uv pip install` like `--script <path>` instead of `-r <path>`).

Reply:
Would `--with-script <script>` mean we're just using the requirements from `<script>` or that we're running it? For the former, I think something like `--with-script-requirements` would be less confusing (I'd read `uv run --with-script` to mean we're running that script).

Reply:
Given every other format has a specific filename(?) we could just have a rule that if none match we assume it's a pep723 script. I like that at least `--with-requirements` seems to more imply the script is being invoked, so if we want a separate argument it would be nice to preserve that (`--script` or `--with-script` is further from that).

Reply:
(oh whoops typo: meant to say `--with-requirements` implies the script *isn't* being invoked)